macros/upload_artifacts_v2.sql

```diff
@@ -0,0 +1,218 @@
+{% endset %}
+
+{% set results_query %}
+    begin;
+    insert into {{ src_results }}
+        with raw_data as (
+
+            select
+                run_results.$1:metadata as metadata,
+                run_results.$1:args as args,
+                run_results.$1:elapsed_time::float as elapsed_time
+            from  @{{ artifact_stage }} as run_results
+
+        )
+        
+        select
+            metadata:invocation_id::string as command_invocation_id,
+            metadata:env:DBT_CLOUD_RUN_ID::int as dbt_cloud_run_id,
+            sha2_hex(coalesce(dbt_cloud_run_id::string, command_invocation_id::string), 256) as artifact_run_id,
+            metadata:generated_at::timestamp_ntz as artifact_generated_at,
+            metadata:dbt_version::string as dbt_version,
+            metadata:env as env,
+            elapsed_time,
+            args:which::string as execution_command,
+            coalesce(args:full_refresh, 'false')::boolean as was_full_refresh,
+            args:models as selected_models,
+            args:target::string as target,
+            metadata,
+            args
+        from raw_data;
+    commit;
+{% endset %}
+
+{% set result_nodes_query %}
+    begin;
+    insert into {{ src_result_nodes }}
+        with raw_data as (
+
+            select
+                run_results.$1:metadata as metadata,
+                run_results.$1:args as args,
+                run_results.$1:results as results
+            from  @{{ artifact_stage }} as run_results
+
+        )
+        
+        select
+            metadata:invocation_id::string as command_invocation_id,
+            metadata:env:DBT_CLOUD_RUN_ID::int as dbt_cloud_run_id,
+            sha2_hex(coalesce(dbt_cloud_run_id::string, command_invocation_id::string), 256) as artifact_run_id,
+            metadata:generated_at::timestamp_ntz as artifact_generated_at,
+            result.value:unique_id::string as node_id,
+            split(result.value:thread_id::string, '-')[1]::integer as thread_id,
+            result.value:status::string as status,
+            -- The first item in the timing array is the model-level `compile`
+            result.value:timing[0]:started_at::timestamp_ntz as compile_started_at,
+            -- The second item in the timing array is `execute`.
+            result.value:timing[1]:completed_at::timestamp_ntz as query_completed_at,
+            -- Confusingly, this does not match the delta of the above two timestamps.
+            -- should we calculate it instead?
+            coalesce(result.value:execution_time::float, 0) as total_node_runtime,
+            result.value:adapter_response:rows_affected::int as rows_affected,
+            -- Future proof by also storing the whole json body to use later if we need to.
+            result.value as result_json
+        from raw_data,
+            lateral flatten(input => results) as result;
+    commit;
+{% endset %}
+
+{% set manifest_nodes_query %}
+    begin;
```

Comment:
```suggestion
    begin;
```

Not sure that the begin; and commit; are necessary - the run_query macro handles all that to my knowledge